content/blog/getting-started-with-spark-on-mapr-sandbox.md

```diff
@@ -0,0 +1,217 @@
+                new PairFunction<String, String, Integer>() {
+                    public Tuple2<String, Integer> call(String x) {
+                        return new Tuple2(x, 1);
+                    }
+                }
+        );
+```
+
+**Transform wordOnePairs RDD with reduceByKey**
+
+To count the number of times each word occurs, we combine the values (1) in the wordOnePairs with the same key (word) using [reduceByKey(func)](https://spark.apache.org/docs/1.3.0/programming-guide.html), This transformation will return an RDD of (word, count) pairs where the values for each word are aggregated using the given reduce function func x+y:
+
+![](/img/tutorial_spark7.png)
+
+```Java
+        // reduce add the pairs by key to produce counts
+        JavaPairRDD<String, Integer> counts = wordOnePairs.reduceByKey(
+                new Function2<Integer, Integer, Integer>() {
+                    public Integer call(Integer x, Integer y) {
+                        return x + y;
+                    }
+                }
+        );
+```
+
+**Output with RDD action saveAsTextFile**
+
+Finally, the RDD action saveAsTextFile(path) writes the elements of the dataset as a text file (or set of text files) in the outputFile directory.
+
+```Java
+        String outputFile = args[1];
+        // Save the word count back out to a text file, causing evaluation.
+        counts.saveAsTextFile(outputFile);                
+```
+
+**Example Word Count App in Scala**
+
+Here is the same example in Scala:
+
+![](/img/tutorial_spark8.png)
+
+**Building a Simple Application**
+
+Spark can be linked into applications in either Java, Scala, or Python.\
+Maven is a popular package management tool for Java-based languages that lets you link to libraries in public repositories. In Java and Scala, you give your application a Maven dependency on the spark-core artifact. The current Spark version is 1.6.1 and the Maven coordinates are:
+
+```xml
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-core_2.10</artifactId>
+            <version>1.6.1</version>
+            <scope>provided</scope>
+        </dependency>
+```
+
+You can build with Maven using IDEs like Eclipse or NetBeans, and then copy the JAR file to your MapR Sandbox, or you can install Maven on your sandbox and build from the Linux command line.
+
+![](/img/tutorial_spark9.png)
+
+You can use scp to copy your JAR file to the MapR Sandbox:
+
+![](/img/tutorial_spark10.png)
+
+* See below for an example of using scp from the command line:\
+  use userid: user01 and password: mapr.
+
+  For VMWare use: `$ scp nameoffile.jar user01@ipaddress:/user/user01/.`
+
+  For Virtualbox use: `$ scp -P 2222 nameoffile.jar user01@127.0.0.1:/user/user01/.`
+
+**Running Your Application**
+
+First find the version of Spark on the sandbox with `ls /opt/mapr/spark/` Then you can use the spark commands in the /opt/mapr/spark/spark-version/bin directory.
```

Comment:
Add comma after "First" and a period at the end of the sentence.

![image](https://user-images.githubusercontent.com/77496261/137521016-0b6b5838-bed1-4b94-81b3-ed90bb418b5c.png)

Reply:
done